aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -445,6 +474,30 @@ impl AggregatorData {
         }
     }
 
+    pub fn snapshot(&mut self, id: &AggregatorID) -> AggregatorSnapshotID {
+        let snapshot_id = AggregatorSnapshotID {
+            id: self.generate_id(),
+        };
+        let aggregator = self.aggregators.get(id).expect("Aggregator doesn't exist");
+        self.aggregator_snapshots
+            .insert(snapshot_id, AggregatorSnapshot {
+                value: aggregator.value,
+                state: aggregator.state,
+                limit: aggregator.limit,
+                history: aggregator.history.clone(),
+                base_aggregator: *id,
+            });
+        snapshot_id
+    }
+
+    pub fn read_snapshot(&self, id: AggregatorSnapshotID) -> u128 {
+        let snapshot = self
+            .aggregator_snapshots
+            .get(&id)
+            .expect("AggregatorSnapshot doesn't exist");
+        snapshot.value
```

Comment:
don't you need here to actually do an equivalent of aggregator read, that jumps over to base_aggregator to continue ?